File: jlox/plox/native_print.py
from plox.lox_callable import LoxCallable


class NativePrint(LoxCallable):
    def arity(self) -> int:
        return 1

    def call(self, interpreter, arguments: list):
        print(interpreter.stringify(arguments[0]))
        return None

    def __str__(self) -> str:
        return "<native fn>"

File: jlox/lox/lox/NativePrint.java
package lox;

import java.util.List;

class NativePrint implements LoxCallable {
    @Override
    public int arity() {
        return 1;
    }

    @Override
    public Object call(Interpreter interpreter, List<Object> arguments) {
        System.out.println(interpreter.stringify(arguments.get(0)));
        return null;
    }

    @Override
    public String toString() {
        return "<native fn print>";
    }

}
